python/lsst/meas/extensions/photometryKron/kronLib.i

```diff
@@ -37,15 +37,39 @@ Interface to Kron magnitudes
 #include "lsst/afw/detection/Footprint.h"
 #include "lsst/afw/image/Utils.h"
 #include "lsst/afw/geom/Point.h"
-#include "lsst/meas/algorithms.h"
-#include "lsst/meas/extensions/photometryKron.h"
+#include "lsst/afw/math.h"
+#include "lsst/afw/image.h"
+#include "lsst/afw/detection.h"
+#include "lsst/afw/table.h"
+#include "lsst/meas/base.h"
+#define PY_ARRAY_UNIQUE_SYMBOL LSST_MEAS_BASE_NUMPY_ARRAY_API
```

Comment:
So if I did need these converters, is the right way to declare the define
and all the other stuff myself, or is there something I should be including
or importing from meas_base?

On Wed, Feb 18, 2015 at 7:00 PM, Jim Bosch <notifications@github.com> wrote:

> In python/lsst/meas/extensions/photometryKron/kronLib.i
> <https://github.com/lsst/meas_extensions_photometryKron/pull/1#discussion_r24963092>
> :
>
> > @@ -37,15 +37,39 @@ Interface to Kron magnitudes
> >  #include "lsst/afw/detection/Footprint.h"
> >  #include "lsst/afw/image/Utils.h"
> >  #include "lsst/afw/geom/Point.h"
> > -#include "lsst/meas/algorithms.h"
> > -#include "lsst/meas/extensions/photometryKron.h"
> > +#include "lsst/afw/math.h"
> > +#include "lsst/afw/image.h"
> > +#include "lsst/afw/detection.h"
> > +#include "lsst/afw/table.h"
> > +#include "lsst/meas/base.h"
> > +#define PY_ARRAY_UNIQUE_SYMBOL LSST_MEAS_BASE_NUMPY_ARRAY_API
>
> A line like this is needed any time you want to use one of ndarrays
> converters to/from NumPy arrays, including those for Eigen objects. It
> looks like you're declaring at least one such converter below (the call to
> %declareNumPyConverters(...). My guess is that you don't actually need
> that here, since I don't think the Kron code ever measures a centroid, and
> hence you could remove this line and all the others that include ndarray or
> NumPy headers, and the %init block that initializes the NumPy library.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lsst/meas_extensions_photometryKron/pull/1/files#r24963092>
> .
>